code/iaas/lb-logic/src/main/java/io/cattle/iaas/lb/service/LoadBalancerService.java

```diff
@@ -0,0 +1,24 @@
+package io.cattle.iaas.lb.service;
+
+import io.cattle.platform.core.model.LoadBalancer;
+import io.cattle.platform.core.model.LoadBalancerConfig;
+
+public interface LoadBalancerService {
+
+    void addHostToLoadBalancer(LoadBalancer lb, long hostId);
+
+    void removeHostFromLoadBalancer(LoadBalancer lb, long hostId);
+
+    void addListenerToConfig(LoadBalancerConfig config, long listenerId);
+
+    void removeListenerFromConfig(LoadBalancerConfig config, long listenerId);
+
+    void addTargetToLoadBalancer(LoadBalancer lb, long instanceId);
+
+    void addTargetToLoadBalancer(LoadBalancer lb, String ipAddress);
```

Comment:
Might want to rename to 'addTargetIpToLoadBalancer'.  I think it's a little risky to have the same name even though the method signature's different.  Plus, it'll match the removeTargetIpFromLoadBalancer() below